app/code/Magento/CatalogSearch/Model/Search/FilterMapper/StockJoinProviderInterface.php

```diff
@@ -0,0 +1,25 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+namespace Magento\CatalogSearch\Model\Search\FilterMapper;
+
+use Magento\Framework\DB\Select;
+
+/**
+ * SPI Interface to change the stock join
+ */
+interface StockJoinProviderInterface
+{
+
+    /**
+     * Add stock join to the given select.
+     *
+     * @param Select $select
+     * @param $alias
+     * @return void
+     */
+    public function add(Select $select, $alias);
```

Comment:
why do we need the interface which adds JOIN to some Select object? 

Why it's not enough just to introduce table resolver, which returns stock index table based on context ?